### ISSUE TITLE
[Bug]: Some L0 flushed segments are left over and the L0 compaction is not triggered

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#30098]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20240117-91aa81b4-amd64
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka): pulsar  
- SDK version(e.g. pymilvus v2.0.0rc2):  pymilvus 2.4.0rc19
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. deploy milvus and enableLevelZero=true
```
 77   config:
 78     dataCoord:
 79       enableActiveStandby: true
 80       segment:
 81         enableLevelZero: true
 82     indexCoord:
 83       enableActiveStandby: true
 84     log:
 85       level: debug
 86     queryCoord:
 87       enableActiveStandby: true
 88     rootCoord:
 89       enableActiveStandby: true
```
2. create collection with 2 shards ->  build index -> insert 10m-128d data -> flush -> index again -> load
3. concurrent requests from 2024-01-17 17:42:49 to 2024-01-17 23:42:50
  - upsert: pk increase from 0 auto
  - flush
  - search
4. during concurrent, apply chaos at 2024-01-17 19:32 and kill all 3 querynodes every 2 minutes. After 10 minutes, remove chaos
5. There are some flushed-L0 segments left after the test is completed. I'm guessing some L0 compaction plan trigger failed
![image](https://github.com/milvus-io/milvus/assets/27288593/7373e2ad-eb37-4e5f-b484-218462486111)
![image](https://github.com/milvus-io/milvus/assets/27288593/5fc6870a-c36e-481d-8b9f-e1e8c633c9ea)

### Milvus Log

- argo: [chaos-level-upsert-1a](https://argo-workflows.zilliz.cc/archived-workflows/qa/821e7246-f33e-4197-bfc4-46f2728973b3?nodeId=chaos-level-upsert-1a)
- grafana: [metrics of chaos-level-upsert-1a](https://grafana-4am.zilliz.cc/d/uLf5cJ3Ga/milvus2-0?orgId=1&var-datasource=prometheus&var-cluster=&var-namespace=qa-milvus&var-instance=chaos-level-upsert-1a&var-collection=All&var-app_name=milvus&from=1705512720000&to=1705546800000)
- pods:
```
chaos-level-upsert-1a-milvus-datanode-d79dc948f-7ffpn             1/1     Running     0                14h
chaos-level-upsert-1a-milvus-datanode-d79dc948f-kjx8l             1/1     Running     0                14h
chaos-level-upsert-1a-milvus-indexnode-6fb9657bd8-cn5d2           1/1     Running     0                14h
chaos-level-upsert-1a-milvus-indexnode-6fb9657bd8-z8n88           1/1     Running     0                14h
chaos-level-upsert-1a-milvus-mixcoord-6676cf46bd-r97t9            1/1     Running     0                14h
chaos-level-upsert-1a-milvus-proxy-744f49f648-qm4qw               1/1     Running     0                14h
chaos-level-upsert-1a-milvus-querynode-7d44f8b54f-2wq6p           1/1     Running     0                12h
chaos-level-upsert-1a-milvus-querynode-7d44f8b54f-m4nts           1/1     Running     0                12h
chaos-level-upsert-1a-milvus-querynode-7d44f8b54f-zqpsc           1/1     Running     0                12h
```

**Comment by xiaofan-luan** [comment]
is this still a issue?

**Comment by ThreadDao** [comment]
waiting for pr merged

**Comment by XuanYang-cn** [comment]
/assign @ThreadDao 
/unassign
Please help verify

**Comment by ThreadDao** [comment]
fixed `master-20240311-a99143dd-amd64`
![image](https://github.com/milvus-io/milvus/assets/27288593/44cca3f2-d534-4296-a6f5-1c72baa7a982)
